csrc/custom_all_reduce.cuh

```diff
@@ -237,7 +305,8 @@ DINLINE P* get_tmp_buf(Signal* sg) {
 template <typename T, int ngpus>
 __global__ void __launch_bounds__(512, 1)
     cross_device_reduce_2stage(RankData* _dp, RankSignals sg, Signal* self_sg,
-                               T* __restrict__ result, int rank, int size) {
+                               T* __restrict__ result, int rank, int size,
+                               int compression_factor = 1) {
```

Comment:
Nit: It looks like you aren't using the compression factor yet? If not, let's hold off adding it until it's used to eliminate any confusion in the short-term.